frontend/www/js/omegaup/arena/arena.ts

```diff
@@ -1881,17 +1882,75 @@ export class Arena {
   }
 
   detectShowRun(): void {
+    const self = this;
     const showRunRegex = /.*\/show-run:([a-fA-F0-9]+)/;
     const showRunMatch = window.location.hash.match(showRunRegex);
     if (!showRunMatch) {
       return;
     }
     $('#overlay form:not([data-run-submit])').hide();
     $('#overlay').show();
-    api.Run.details({ run_alias: showRunMatch[1] })
+    const guid = showRunMatch[1];
+    api.Run.details({ run_alias: guid })
       .then(time.remoteTimeAdapter)
       .then((data) => {
-        this.displayRunDetails(showRunMatch[1], data);
+        if (
+          data.show_diff === 'none' ||
+          (self.options.contestAlias && self.options.contestAlias !== 'admin')
+        ) {
+          self.displayRunDetails(guid, data);
+          return;
+        } else {
+          fetch(`/api/run/download/run_alias/${guid}/show_diff/true/`)
+            .then((response) => {
+              if (response.status === 200 || response.status === 0) {
+                return Promise.resolve(response.blob());
+              } else {
+                return Promise.reject(new Error(response.statusText));
+              }
+            })
+            .then(JSZip.loadAsync)
+            .then((zip: JSZip) => {
+              const result: {
+                cases: string[];
+                promises: any;
+              } = { cases: [], promises: [] };
+              zip.forEach((relativePath, zipEntry) => {
+                if (
+                  relativePath.split('.').pop() === 'out' &&
+                  relativePath.indexOf('/') === -1
+                ) {
+                  if (
+                    data.show_diff === 'examples' &&
+                    relativePath.indexOf('sample') === -1
+                  ) {
+                    return;
+                  }
+                  const pos = relativePath.lastIndexOf('.');
+                  const basename = relativePath.substring(0, pos);
+                  result.cases.push(basename);
+                  result.promises.push(zip.file(zipEntry.name).async('text'));
```

Comment:
qué pasaría si aquí en vez de un arreglo de promesas, se hace todo mediante `await`?

```suggestion
                  data.cases[basename] = await zip.file(zipEntry.name).async('text');
```

el lambda de L1918 quizás necesite un `async`.

Reply:
Es necesario que lo haga con promesas, ya que de otra forma `Promise.allSettled` no recibe nada. A menos que no lo esté haciendo correctamente.

Reply:
bah, me acabo de dar cuenta que `zip.forEach` no se lleva bien con callbacks asíncronos. tons dejemos esto así.